torch/_subclasses/functional_tensor.py

```diff
@@ -316,6 +336,10 @@ def __init__(self, pre_dispatch=False, export=False, _allow_token_discovery=Fals
         # discovery. This flag distinguishes between the two stages.
         self._allow_token_discovery = _allow_token_discovery
 
+        self._storage_to_base: Dict[
+            torch.storage.UntypedStorage, Optional[weakref.ReferenceType[FunctionalTensor]]
```

Comment:
As discussed offline we probably want WeakKeyDictionary[UntypedStorage, FunctionalTensor]